src/components/PostsList.tsx

```diff
@@ -1,85 +1,59 @@
-          </td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">18</td>
-
-          <td data-cy="PostTitle">
-            voluptate et itaque vero tempora molestiae
-          </td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link"
-            >
-              Close
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">19</td>
-          <td data-cy="PostTitle">adipisci placeat illum aut reiciendis qui</td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-
-        <tr data-cy="Post">
-          <td data-cy="PostId">20</td>
-          <td data-cy="PostTitle">doloribus ad provident suscipit at</td>
-
-          <td className="has-text-right is-vcentered">
-            <button
-              type="button"
-              data-cy="PostButton"
-              className="button is-link is-light"
-            >
-              Open
-            </button>
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </div>
-);
+import React, { Dispatch, SetStateAction } from 'react';
+import { Post } from '../types/Post';
+import { PostItem } from './PostItem';
+
+type Props = {
+  posts: Post[];
+  selectedPost: Post | null;
+  setSelectedPost: Dispatch<SetStateAction<Post | null>>;
+  setAddError: Dispatch<SetStateAction<boolean>>;
+  setDeleteError: Dispatch<SetStateAction<boolean>>;
+};
+
+export const PostsList: React.FC<Props> = ({
+  posts,
+  selectedPost,
+  setSelectedPost,
+  setAddError,
+  setDeleteError,
+}) => {
+  const handlePostSelect = (post: Post) => {
+    setAddError(false);
+    setDeleteError(false);
+
+    if (post.id === selectedPost?.id) {
+      setSelectedPost(null);
+
+      return;
+    }
+
+    setSelectedPost(post);
+  };
+
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
+
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {posts.map(post => (
+            <PostItem
+              key={post.id}
+              post={post}
+              selectedPostId={!selectedPost ? 0 : selectedPost.id}
```

Comment:
```suggestion
              selectedPostId={selectedPost}
```
it might cover a case with `null` value